### PR TITLE
159 decompose oecore procedures to smaller

### PR DESCRIPTION
**Pull request by Einingis**

### CODE REVIEW

File: resources/oe/src/oeCore.i [hunk 1]
```diff
@@ -1,3 +1,19 @@
+FUNCTION GET_TABLE_TYPE RETURNS CHARACTER (qhFile AS WIDGET-HANDLE):
```
[the remark this answers is not in the document]
Comment (Einingis) [reply resources/oe/src/oeCore.i:1]: it work if we pass just filename and number

[the remark this answers is not in the document]
Comment (Knijus) [reply resources/oe/src/oeCore.i:1]: https://github.com/BalticAmadeus/ProBro/pull/282#discussion_r1130711349 so I guess try expanding this function and return json object with tableRow

File: resources/oe/src/oeCore.i [hunk 2]
```diff
@@ -74,69 +92,53 @@ PROCEDURE LOCAL_GET_TABLES:
...
-		THEN jsonTableRow:Add("tableType", "UserTable").
-		ELSE IF qh:GET-BUFFER-HANDLE(1)::_file-number > -80 AND qh:GET-BUFFER-HANDLE(1)::_file-number < 0
-		THEN jsonTableRow:Add("tableType", "SchemaTable").
-		ELSE IF qh:GET-BUFFER-HANDLE(1)::_file-number < -16384 
-		THEN jsonTableRow:Add("tableType", "VirtualSystem").
-		ELSE IF qh:GET-BUFFER-HANDLE(1)::_file-number >= -16384 AND qh:GET-BUFFER-HANDLE(1)::_file-number <= -80
-		THEN jsonTableRow:Add("tableType", "OtherTables").
-		ELSE NEXT.
-	
+		jsonTableRow:Add("name", qhFile:GET-BUFFER-HANDLE(1)::_file-name).
```
[the remark this answers is not in the document]
Comment (Einingis) [reply resources/oe/src/oeCore.i:109]: We can do that but then checking if we should skip some rows becomes awkward so I think we can leave it like this.

[the remark this answers is not in the document]
Comment (Knijus) [reply resources/oe/src/oeCore.i:109]: What kind of tables should be skipped?

**Comment by PauliusKu** [comment]
@Knijus approve if ok
@Einingis solve merge conflict

File: resources/oe/src/oeCore.i [hunk 3]
```diff
@@ -1,708 +1,802 @@
...
+END FUNCTION.
+
+FUNCTION GET_MODE RETURNS CHARACTER ():
+    IF inputObject:GetJsonObject("params"):has("mode") THEN DO:
+        RETURN inputObject:GetJsonObject("params"):GetCharacter("mode").
+    END.
+    ELSE DO:
+        RETURN "DATA".
+    END.
+END FUNCTION.
```
Comment (Knijus) [inline resources/oe/src/oeCore.i:406]: When resolving conflict: 
```
FUNCTION GET_MODE RETURNS CHARACTER ():
```suggestion
FUNCTION GET_MODE RETURNS CHARACTER ():
    IF inputObject:GetJsonObject("params"):has("mode") THEN DO:
        RETURN inputObject:GetJsonObject("params"):GetCharacter("mode").
    END.
    ELSE DO:
        RETURN "DATA".
    END.
END FUNCTION.
```
```suggestion
FUNCTION GET_MODE RETURNS CHARACTER ():
    IF inputObject:GetJsonObject("params"):has("mode") THEN DO:
        RETURN inputObject:GetJsonObject("params"):GetCharacter("mode").
    END.
    ELSE DO:
        RETURN "DATA".
    END.
END FUNCTION.
```
    IF NOT inputObject:GetJsonObject("params"):has("mode") THEN RETURN "DATA".
    RETURN inputObject:GetJsonObject("params"):GetCharacter("mode").
END FUNCTION.
```

**Pull request merged**